client.go

```diff
@@ -1965,35 +1967,56 @@ func (c *Client) SetTrace(t bool) *Client {
 	return c
 }
 
-// EnableGenerateCurlOnDebug method enables the generation of CURL commands in the debug log.
-// It works in conjunction with debug mode.
+// EnableGenerateCurlCmd method enables the generation of curl command at the
+// client instance level.
+//
+// By default, Resty does not log the curl command in the debug log since it has the potential
+// to leak sensitive data unless explicitly enabled via [Client.SetDebugLogCurlCmd].
 //
 // NOTE: Use with care.
-//   - Potential to leak sensitive data from [Request] and [Response] in the debug log.
-//   - Beware of memory usage since the request body is reread.
-func (c *Client) EnableGenerateCurlOnDebug() *Client {
-	c.SetGenerateCurlOnDebug(true)
+//   - Potential to leak sensitive data from [Request] and [Response] in the debug log
+//     when the debug log option is enabled.
+//   - Additional memory usage since the request body was reread.
+//   - curl body is not generated for [io.Reader] and multipart request flow.
+func (c *Client) EnableGenerateCurlCmd() *Client {
+	c.SetGenerateCurlCmd(true)
 	return c
 }
 
-// DisableGenerateCurlOnDebug method disables the option set by [Client.EnableGenerateCurlOnDebug].
-func (c *Client) DisableGenerateCurlOnDebug() *Client {
-	c.SetGenerateCurlOnDebug(false)
+// DisableGenerateCurlCmd method disables the option set by [Client.EnableGenerateCurlCmd] or
+// [Client.SetGenerateCurlCmd].
+func (c *Client) DisableGenerateCurlCmd() *Client {
+	c.SetGenerateCurlCmd(false)
 	return c
 }
 
-// SetGenerateCurlOnDebug method is used to turn on/off the generate CURL command in debug mode
-// at the client instance level. It works in conjunction with debug mode.
+// SetGenerateCurlCmd method is used to turn on/off the generate curl command at the
+// client instance level.
+//
+// By default, Resty does not log the curl command in the debug log since it has the potential
+// to leak sensitive data unless explicitly enabled via [Client.SetDebugLogCurlCmd].
 //
 // NOTE: Use with care.
-//   - Potential to leak sensitive data from [Request] and [Response] in the debug log.
-//   - Beware of memory usage since the request body is reread.
+//   - Potential to leak sensitive data from [Request] and [Response] in the debug log
+//     when the debug log option is enabled.
+//   - Additional memory usage since the request body was reread.
+//   - curl body is not generated for [io.Reader] and multipart request flow.
+//
+// It can be overridden at the request level; see [Request.SetGenerateCurlCmd]
+func (c *Client) SetGenerateCurlCmd(b bool) *Client {
```

Comment:
This one shouldn't be exported.

Otherwise, you had no reason to add Enable and Disable one,no?

Or maybe I'm missing something

Reply:
I kept two kinds of usage variation: Resty users choose one approach based on their need/preference. You may see this in nature methods for frequently used ones, e.g. Debug, Trace, curl, and retry default conditions.